### DISCUSSION TITLE
Networking General 📡

### DISCUSSION THREAD

**Discussion by 4c3e** [markqvist/Reticulum#27] [General]
Hello! If you are interested in hosting a public RNS node, please feel free to comment about it in this general :smile: 

Please proceed with caution if you are publicly posting your IP or geographic location for radio. A safer alternative is to host a RNS node over [I2P](https://geti2p.net/en/), or leave contact info for those interested.

**Comment by 4c3e** [comment]
~~Will update this comment with this node's info (and hopefully a guide) when it's live.~~

**Reply by 4c3e** [reply]
Running a new LXMF node, here is the config:

```
  [[4c3ePiNode]]
     type = I2PInterface
     interface_enabled = yes
     peers = re2ct667tpclusj4neu2yz6i2kqp23tnbldbiexiohund2gw2yrq.b32.i2p
```

Please feel free to add to your config to add connectivity to the testnet :)

If you already have an I2P interface set up, you can simply add `,re2ct667tpclusj4neu2yz6i2kqp23tnbldbiexiohund2gw2yrq.b32.i2p` (note the comma) to the "peers" line in your config.

**Reply by markqvist** [reply]
I will be connecting a node or two to this endpoint now. Thanks for supplying it!

**Comment by markqvist** [comment]
Great idea! For nodes that intent to stay up long-term, I can add them to a list of public nodes in the repo documentation.

**Comment by markqvist** [comment]
> Hello! If you are interested in hosting a public RNS node, please feel free to comment about it in this general smile
> 
> Please proceed with caution if you are publicly posting your IP or geographic location for radio. A safer alternative is to host a RNS node over [I2P](https://geti2p.net/en/), or leave contact info for those interested.

This is relevant and important. Ideally, we should have a small section in the manual that explains this, along with a very general explanation of hosting publicly accessible nodes over I2P vs "normal" TCP. Is this something you might be up for writing @4c3e?

**Reply by 4c3e** [reply]
Sure! Would it be best to add that info along side the clearnet/i2p interface definitions in the manual?

**Reply by markqvist** [reply]
Ideally I'd like to keep the interface documentation as clean and focused on the configuration parameters as possible. What about inserting a new section after [Creating a Network With Reticulum](https://markqvist.github.io/Reticulum/manual/gettingstartedfast.html#creating-a-network-with-reticulum) section of the Getting Started chapter. That allow a bit more paragraph space to explain the different approaches, and when it could be valuable to use I2P over clearnet interfaces. Then we can just link to that section from each interface section with a small note. What do you think?

**Reply by 4c3e** [reply]
Sounds good. I'll start writing something up, and make a PR when I think it's ready for review.

**Comment by ZS6GJM** [comment]
My node, **node-0.rns-za.arza.org.za** has been configure for long-term use and the first one to come online in South Africa AFAIK - primarily made available for Amateur Radio operators in South-Africa. I have a website available (still a work-in-progress) as well as a registration system (with statistics, work-in-progress) where fellow AR operators can register in order to "hook" onto the node-0.rns-za.arza.org.za and establish communications. Once the system coding is complete, my intention is to get in contact with all the Amateur Radio clubs in South Africa to start an awareness campaign and get people to hook up to the system.

**Reply by 4c3e** [reply]
Awesome! For always on nodes it might be worth it to follow the [Nomadnet Docker Daemon](https://github.com/markqvist/NomadNet/pull/5) pull request, I'm not too familiar with Docker but it might offer some features for always-on nodes.

**Reply by markqvist** [reply]
It's not quite ready yet, since it is still waiting for me to actually add the UI-less daemon mode to nomadnet. But it will be there soon :)

**Comment by arizonajon** [comment]
I can put up one here at my location in PHX. I currently am testing with an RNode on 900 MHz, and have achieved ranges of over 10 km. I have some interest in very low-power remote sensing things out in the desert, and having a way for them to trickle data back would be cool.

**Reply by markqvist** [reply]
That would be great! You should have a look at the new interface options introduced in version 0.3.5. You can now set interfaces to "access point" mode, and also create authenticated or named networks. It is all documented in the [interfaces section of the manual](https://markqvist.github.io/Reticulum/manual/interfaces.html). Let me know if you have any questions regarding how to do things in those regards.
